src/Microsoft.OData.Edm/Csdl/CsdlReader.Json.cs

```diff
@@ -0,0 +1,201 @@
+            if (!TryParse(ref reader, out model, out parseErrors))
+            {
+                throw new EdmParseException(parseErrors);
+            }
+
+            return model;
+        }
+
+        /// <summary>
+        /// Returns an IEdmModel for the given CSDL-JSON artifact.
+        /// </summary>
+        /// <param name="reader">The given JSON reader containing the CSDL artifact.</param>
+        /// <param name="settings">The CSDL-JSON reader settings.</param>
+        /// <returns>The model generated by parsing.</returns>
+        public static IEdmModel Parse(ref Utf8JsonReader reader, CsdlJsonReaderSettings settings)
+        {
+            IEdmModel model;
+            IEnumerable<EdmError> parseErrors;
+            if (!TryParse(ref reader, settings, out model, out parseErrors))
+            {
+                throw new EdmParseException(parseErrors);
+            }
+
+            return model;
+        }
+
+        /// <summary>
+        /// Load and parse the referenced model but ignored any further referenced model.
+        /// </summary>
+        /// <param name="csdlModel">The main CSDL model.</param>
+        /// <param name="context">The parser context.</param>
+        /// <returns>A list of CsdlModel (no semantics) of the referenced models.</returns>
+        private static List<CsdlModel> LoadReferencedCsdl(CsdlModel csdlModel, JsonParserContext context)
+        {
+            List<CsdlModel> referencedAstModels = new List<CsdlModel>();
+            if (context.Settings.JsonReaderFactory == null)
+            {
+                // don't try to load CSDL-JSON doc, but this.edmReferences's namespace-alias need to be used later.
+                return referencedAstModels;
+            }
+
+            foreach (var edmReference in csdlModel.CurrentModelReferences)
+            {
+                // If nothing included, why does it exist?
+                if (!edmReference.Includes.Any() && !edmReference.IncludeAnnotations.Any())
+                {
+                    continue;
+                }
+
+                // Skip the built-in vocabulary annotation model
+                if (edmReference.Uri != null && (edmReference.Uri.OriginalString.EndsWith("/Org.OData.Core.V1.json", StringComparison.Ordinal) ||
+                    edmReference.Uri.OriginalString.EndsWith("/Org.OData.Capabilities.V1.json", StringComparison.Ordinal) ||
+                    edmReference.Uri.OriginalString.EndsWith("/Org.OData.Authorization.V1.json", StringComparison.Ordinal) ||
+                    edmReference.Uri.OriginalString.EndsWith("/Org.OData.Validation.V1.json", StringComparison.Ordinal) ||
+                    edmReference.Uri.OriginalString.EndsWith("/Org.OData.Community.V1.json", StringComparison.Ordinal) ||
+                    edmReference.Uri.OriginalString.EndsWith("/OData.Community.Keys.V1.json", StringComparison.Ordinal)))
+                {
+                    continue;
+                }

```

Comment:
I would think that, if includeDefaultVocabularies is false, we should load any explicitly referenced vocabularies

Reply:
Yes.  "includeDefaultVocabularies" is used in "SemanticEdmModel",  in "SemanticEdmModel" constructor. 

here, we just skip loading the built-in vocabularies model. Because, the built-in vocabularies are loaded using it's own process.

---
In reply to: [429393994](https://github.com/OData/odata.net/pull/1775#discussion_r429393994) [](ancestors = 429393994)